e2e/app-with-domain/app_with_domain_test.go

```diff
@@ -102,6 +102,86 @@ var _ = Describe("App With Domain", func() {
 				Expect(err).NotTo(HaveOccurred())
 			}
 		})
+
+		It("should create environment manifests", func() {
+			Expect("./copilot/environments/test/manifest.yml").Should(BeAnExistingFile())
+			Expect("./copilot/environments/prod/manifest.yml").Should(BeAnExistingFile())
+		})
+
+		It("should show both environments in env ls", func() {
+			envList, envListError := cli.EnvList(appName)
+			Expect(envListError).NotTo(HaveOccurred())
+			Expect(len(envList.Envs)).To(Equal(2))
+		})
+
+		It("should show only bootstrap resources in env show", func() {
+			testEnvShowOutput, testEnvShowError := cli.EnvShow(&client.EnvShowRequest{
+				AppName: appName,
+				EnvName: "test",
+			})
+			prodEnvShowOutput, prodEnvShowError := cli.EnvShow(&client.EnvShowRequest{
+				AppName: appName,
+				EnvName: "prod",
+			})
+			Expect(testEnvShowError).NotTo(HaveOccurred())
+			Expect(prodEnvShowError).NotTo(HaveOccurred())
+
+			// Contains only bootstrap resources - two IAM roles.
+			Expect(len(testEnvShowOutput.Resources)).To(Equal(2))
+			Expect(len(prodEnvShowOutput.Resources)).To(Equal(2))
+		})
+	})
+
+	Context("when deploying the environments", func() {
+		fatalErrors := make(chan error)
+		wgDone := make(chan bool)
+		It("env deploy should succeed for deploying test and prod environments", func() {
```

Comment:
Can we separate the wait group function (takes env init or env deploy func as a param) so that it can be share by both create and deploy?

Reply:
talked offline!